src/common/editFileDialog.component.tsx

```diff
@@ -0,0 +1,207 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormHelperText,
+  Grid,
+  InputAdornment,
+  TextField,
+} from '@mui/material';
+import { useForm } from 'react-hook-form';
+
+import React from 'react';
+
+import { UseMutationResult } from '@tanstack/react-query';
+import { AxiosError } from 'axios';
+import { APIImage, ObjectFilePatch } from '../api/api.types';
+import { FileSchemaPatch } from '../form.schemas';
+import handleIMS_APIError from '../handleIMS_APIError';
+import { getNameAndExtension } from '../utils';
+
+export interface BaseFileDialogProps {
+  open: boolean;
+  onClose: () => void;
+  fileType: 'Image' | 'Attachment';
+}
+
+export interface ImageDialogProps extends BaseFileDialogProps {
+  fileType: 'Image';
+  selectedFile: APIImage;
+  usePatchFile: () => UseMutationResult<
+    APIImage,
+    AxiosError,
+    { id: string; fileMetadata: ObjectFilePatch }
+  >;
+}
+
+export type FileDialogProps = ImageDialogProps;
+
+const EditFileDialog = (props: FileDialogProps) => {
+  const { open, onClose, selectedFile, fileType, usePatchFile } = props;
+
+  const { mutateAsync: patchFile, isPending: isEditPending } = usePatchFile();
+
+  const [initialName, extension] = getNameAndExtension(selectedFile.file_name);
+
+  const initialFile = React.useMemo<ObjectFilePatch>(() => {
+    return {
+      ...selectedFile,
+      file_name: initialName,
+    };
+  }, [selectedFile, initialName]);
+
+  const {
+    handleSubmit,
+    register,
+    formState: { errors },
+    watch,
+    setError,
+    clearErrors,
+    reset,
+  } = useForm<ObjectFilePatch>({
+    resolver: zodResolver(FileSchemaPatch()),
```

Comment:
The FileSchemaPatch() implementation here is unnecessary because FileSchemaPatch doesn't require any parameters or dynamic evaluation. Since it’s not a function or callable that needs to be invoked, it’s better to pass it directly to the resolver as `zodResolver(FileSchemaPatch)`.